src/lib/libXbSymbolDatabase.c

```diff
@@ -1931,6 +1943,119 @@ static bool manual_scan_section_dsound(iXbSymbolContext* pContext,
+                    break;
+                }
+            }
+
+            xXbAddr = (xbaddr)(uintptr_t)LocateFunctionCast(pContext, iLibraryType, "MU_Init", list_MU_Init[i].version,
+                                                            list_MU_Init[i].sig, pSection);
+
+            // If match is found then register symbol.
+            if (xXbAddr != 0) {
+
+                if (pLibrary->build_version < list_MU_Init[i].version) {
+
+                    output_message_format(&pContext->output, XB_OUTPUT_MESSAGE_WARN,
+                                          "OOVPA signature is too high for [%hu] %s!",
+                                          list_MU_Init[i].version, "MU_Init");
+                }
+
+                internal_RegisterSymbol(pContext, pLibrarySession, XREF_MU_Init, list_MU_Init[i].version,
+                                        "MU_Init", xXbAddr);
+
+                if (pContext->one_time_scan) {
+                    break;
+                }
+            }
+        }
+    }
+
+    if (!internal_IsXRefAddrValid(pContext->xref_database[XREF_MU_Init])) {
+        // If not found, skip the rest of the scan.
+        return false;
+    }
+
+
+    // Scan for IUsbInit::GetMaxDeviceTypeCount function.
+    if (!internal_IsXRefAddrValid(pContext->xref_database[XREF_IUsbInit_GetMaxDeviceTypeCount])) {
+        xXbAddr = (xbaddr)(uintptr_t)LocateFunctionCast(pContext, iLibraryType, "IUsbInit_GetMaxDeviceTypeCount", 3911,
+                                                        &IUsbInit_GetMaxDeviceTypeCount_3911, pSection);
+
+        // If not found, skip the rest of the scan.
+        if (xXbAddr == 0) {
+            return false;
+        }
+
+        internal_RegisterSymbol(pContext, pLibrarySession, XREF_IUsbInit_GetMaxDeviceTypeCount, 3911,
+                                "IUsbInit_GetMaxDeviceTypeCount", xXbAddr);
+    }
+
+
+    xXbAddr = 0;
+    // TODO: Move below into separate function.
+    // search all of the buffer memory range
+    for (memptr_t cur = (memptr_t)virt_start_relative + pContext->xref_database[XREF_MU_Init]; cur < buffer_upper; cur++) {
+        // We are looking for "leave; ret 0x04;" for end of function then stop searching.
+        if (cur[0] == 0xC9 && cur[1] == 0xC2 && cur[2] == 0x04) {
+            output_message_format(&pContext->output, XB_OUTPUT_MESSAGE_ERROR, "Could not find g_XapiMountedMUs from MU_Init!");
+            return false;
+        }
+
+        // Look for possible push and call next to each other.
+        if (cur[0] == 0x68 && cur[5] == 0xE8) {
+
+            // check if call is linked to IUsbInit_GetMaxDeviceTypeCount function.
+            xbaddr ActualAddr = *(xbaddr*)(cur + 6);
+            if (MatchXRefAddr(cur + 6, virt_start_relative, pContext->xref_database[XREF_IUsbInit_GetMaxDeviceTypeCount])) {
+                // this is where g_XapiMountedMUs hardcode address reside in.
+                xXbAddr = *(xbaddr*)(cur + 1);
```

Comment:
This is wrong. This is not `g_XapiMountedMUs`, but the MU xpp type. Here is how `MU_Init` looks like in the game Phantasy Star Online
![Capture](https://user-images.githubusercontent.com/45463469/122540271-c35f3d00-d028-11eb-80df-8cdeb47f8527.PNG)
As you can see, the push instruction has the address of an xpp type, not of an ULONG, which is what `g_XapiMountedMUs` is. This global is instead referenced by `XUnmountMU`, and you can check this code [here](https://github.com/ergo720/Cxbx-Reloaded/blob/364fbb610dc6286b1b918cdb117505d0b085963e/src/core/hle/XAPI/Xapi.cpp#L446) to see how I'm deriving it in cxbxr.

Reply:
Thanks, I got confused for a moment when you shown me the offset with `XPP_DEVICE_TYPE` which said otherwise. I made a re-word to `g_DeviceType_MU` and rename the commit. Since `g_XapiMountedMUs` can be easily derive from signatures itself. I only need to manually register xref at end of the scan process.